src/bootstrap/src/core/builder.rs

```diff
@@ -2446,7 +2446,15 @@ impl Cargo {
         cmd_kind: Kind,
     ) -> Cargo {
         let mut cargo = builder.cargo(compiler, mode, source_type, target, cmd_kind);
-        cargo.configure_linker(builder);
+
+        match cmd_kind {
```

Comment:
Maybe we could create `fn needs_linking(&self) -> bool` on `Subcommand`, use it in the two places in this PR and perform an exhaustive match inside of it?

Reply:
That wouldn't work as `Subcommand`s and `cmd_kind: Kind` doesn't have to be the same (e.g., [here](https://github.com/rust-lang/rust/blob/97e72524a34994541810438258115802f8b85543/src/bootstrap/src/core/build_steps/clippy.rs#L255) `x clippy` propagates `Kind::Check` to that function).